testing/ef-tests/src/macros/fork_choice.rs

```diff
@@ -0,0 +1,256 @@
+                                .expect("Failed to parse steps.yaml")
+                        };
+
+                        let anchor_state: BeaconState =
+                            utils::read_ssz_snappy(&case_dir.join("anchor_state.ssz_snappy"))
+                                .expect("Failed to read anchor_state.ssz_snappy");
+                        let anchor_block: BeaconBlock =
+                            utils::read_ssz_snappy(&case_dir.join("anchor_block.ssz_snappy"))
+                                .expect("Failed to read anchor_block.ssz_snappy");
+
+                        let mut store = get_forkchoice_store(anchor_state.clone(), anchor_block.clone())
+                            .expect("get_forkchoice_store failed");
+
+                        for step in steps {
+                            match step {
+                                ForkChoiceStep::Tick(ticks) => {
+                                    let result = on_tick(&mut store, ticks.tick);
+                                    if ticks.valid == Some(false) {
+                                        assert!(result.is_err());
+                                    } else {
+                                        assert!(result.is_ok())
+                                    }
+                                }
+                                ForkChoiceStep::Block(blocks) => {
+                                    let block_path = case_dir.join(format!("{}.ssz_snappy", blocks.block));
+                                    if !block_path.exists() {
+                                        panic!("Test asset not found: {:?}", block_path);
+                                    }
+                                    let block: SignedBeaconBlock = utils::read_ssz_snappy(&block_path)
+                                        .unwrap_or_else(|_| {
+                                            panic!("cannot find test asset (block_{blocks:?}.ssz_snappy)")
+                                        });
+
+                                    if let (Some(blobs), Some(proof)) = (blocks.blobs, blocks.proofs) {
+                                        let blobs_path = case_dir.join(format!("{}.ssz_snappy", blobs));
+                                        let blob: VariableList<Blob, U4096> = utils::read_ssz_snappy(&blobs_path).expect("");
+
+                                        let proof: Vec<KZGProof> = proof
+                                            .into_iter()
+                                            .map(|proof| KZGProof::from_hex(proof).expect("could not get KZGProof"))
+                                            .collect();
+
+                                        let blobs_and_proof = blob.into_iter().zip(proof.into_iter()).map(|(blob, proof)| BlobsAndProofV1 { blob, proof  } ).collect::<Vec<_>>();
+
+                                        store.blobs_and_proofs.insert(block.message.tree_hash_root(), blobs_and_proof);
```

Comment:
```suggestion
                                    if let (Some(blobs), Some(proof)) = (blocks.blobs, blocks.proofs) {
                                        let blobs_path = case_dir.join(format!("{}.ssz_snappy", blobs));
                                        let blob: VariableList<Blob, U4096> = utils::read_ssz_snappy(&blobs_path).expect("");
                                        
                                        for (index, (blob, proof)) in blob.into_iter().zip(proof.into_iter()).enumerate() {
                                            let proof = KZGProof::from_hex(proof).expect("Couldn't decode poof");
                                            store.blobs_and_proofs.insert(BlobIdentifier::new(block.message.tree_hash_root(), index), BlobsAndProofV1 { blob, proof  });

                                        }
```

I think we could do this ^

Reply:
I had to make a few changes because a stack overflow was occurring when i was running a few tests.